idxbroker/wordpress-plugin
# IMPB-1473 Unable to access GMB after saving post

# Pull Requests

🐛 Are you fixing a bug? Yes

### Description of the Change

react's setState operation is asynchronous. this change moves the 'current schedule' logic into a callback for setPosts in getPostsData which prevents the error that was causing the GMB dashboard being inaccessible---the 'posts' variable was being used by React before everything was in place.

### Verification Process

1. With IMPress installed and Listings enabled, connect it to a GMB account.
2. Once connected, create a new GMB post either manually or with a listing.
3. Save the post. Observe that you are no longer able to use the GMB IMPress functionality.
4. Apply fixes in PR. Navigate to Listings > Google My Business. Observe that the GMB settings page loads as expected.

Note that there might be further issues with the GMB setting page, but this PR addresses the specific issue mentioned in IMPB-1473

### Release Notes

fix: restore access to GMB when there's at least one post saved

## Review

Pull Requests must have the sign-off of two other developers and at least one of these must be an IDX Broker team member.
